ocp_resources/rhmi.py

```diff
@@ -8,3 +9,31 @@ class RHMI(NamespacedResource):
     """
 
     api_group = NamespacedResource.ApiGroup.INTEGREATLY_ORG
+
+    def wait_for_install_status_complete(self, timeout):
+        """
+        Wait until RHMI stage status is complete.
+
+        Args:
+            timeout (int): Time to wait for installation status.
+
+        Raises:
+            TimeoutExpiredError: If stage status is not complete.
+
+        """
+        self.logger.info(f"Wait for {self.kind} {self.name} installation status to be {RHMI.Status.COMPLETE}")
+        try:
+            timeout_watcher = TimeoutWatch(timeout=timeout)
+            self.wait(timeout=timeout)
+
+            for sample in TimeoutSampler(
+                wait_timeout=timeout_watcher.remaining_time(),
+                sleep=3,
+                func=lambda: self.instance.status.stage,
+            ):
+                if sample == RHMI.Status.COMPLETE:
+                    return
+
+        except TimeoutExpiredError:
+            self.logger.error(f"after {timeout} seconds, {self.kind} {self.name} stage status is {sample}")
```

Comment:
you need to set `sample` to None before the `try` otherwise it may be used before referenced.